furnace.c

```diff
@@ -162,18 +165,38 @@ command_handler(furnace_context_t* ctx, uint8_t* buffer, void (*feedback)(const
     char msg[16];
     const size_t msg_len = snprintf(msg, sizeof(msg), "temp = %d\r\n", ctx->pilot.des_temp);
     feedback(msg, msg_len);
+  } else if (sscanf(buffer, "log %s %u", &str_arg, &arg) == 2) {
```

Comment:
potentially buffer overflow possible on str_arg. We should not allow reading more characters then there is space in the buffer.